samples: Fixed 54L10 configuration for Matter and Thread (#23467)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

    * :kconfig:option:`CONFIG_CHIP_ICD_ACTIVE_MODE_THRESHOLD` from ``300`` to ``0`` ms for SIT devices.
    * :kconfig:option:`CONFIG_CHIP_ICD_FAST_POLLING_INTERVAL` from ``200`` to ``500`` ms.

>   * The memory layout for the :zephyr:board:`nrf54l15dk/nrf54l10/cpuapp`, as the previous one was invalid and allowed to access memory area out of bounds.

Reviewer: ```suggestion
  * The memory layout for the :zephyr:board:`nrf54l15dk/nrf54l10/cpuapp` board target, as the previous one was invalid and allowed to access memory area out of bounds.
```

>   * The memory layout for the :zephyr:board:`nrf54l15dk/nrf54l10/cpuapp` board target, as the previous one was invalid and allowed to access memory area out of bounds.

Reviewer: ```suggestion
  * The memory layout for the ``nrf54l15dk/nrf54l10/cpuapp`` board target, as the previous one was invalid and allowed to access memory area out of bounds.
```
To overcome the doc build failure.

>   * The memory layout for the :zephyr:board:`nrf54l15dk/nrf54l10/cpuapp` board target, as the previous one was invalid and allowed to access memory area out of bounds.

Reviewer 2: ```suggestion
  * The memory layout for the :ref:`nRF54L15 DK (emulating nRF54L10) <zephyr:nrf54l15dk_nrf54l10>` board target, as the previous one was invalid and allowed to access memory area out of bounds.
```

>     The maximum non-volatile area size was changed from 1022 kB to 1012 kB, the application partition size was decreased by 8 kB and reserved partition was removed.

Reviewer: ```suggestion
    The maximum size of the non-volatile area was changed from 1022 kB to 1012 kB, the application partition size was decreased by 8 kB, and the reserved partition was removed.
```
